### PR TITLE
[Impeller] add basic Impeller+Vulkan support to embedder API.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Works on GLFW example app with no validation errors.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Integration with context vk is a bit messy, since the emebdder expects to provide a lot of objects that context vk was already creating.

**Comment by jwinarske** [comment]
@jonahwilliams awesome 👍. I'll test it out when I'm back in the office; October 14th.

**Comment by jwinarske** [comment]
Also I haven't looked into it yet, but using the compositor interface bypasses some of the embedder API; Fuschia dev mentioned this.  My plans include moving to the compositor interface in the near future.

**Comment by jonahwilliams** [comment]
Sorry I don't follow

File: shell/platform/embedder/embedder.cc [hunk 1]
```diff
@@ -1434,11 +1509,16 @@ CreateEmbedderRenderTarget(
...
     }
     case kFlutterBackingStoreTypeVulkan: {
-      auto skia_surface =
-          MakeSkSurfaceFromBackingStore(context, config, &backing_store.vulkan);
-      render_target = MakeRenderTargetFromSkSurface(
-          backing_store, std::move(skia_surface), collect_callback.Release());
-      break;
+      if (enable_impeller) {
+        FML_LOG(ERROR) << "Unimplemented";
+        break;
```
Comment (chinmaygarde) [inline shell/platform/embedder/embedder.cc:1514]: IIRC, this should return nullptr. That should return the right error code to the embedder. We can file a bug for followup.

**Review by chinmaygarde — APPROVED** [review]

File: shell/platform/embedder/embedder.cc (hunk 1 above)
Comment (jonahwilliams) [reply shell/platform/embedder/embedder.cc:1514]: Breaking here leaves the render_target unique_ptr as nullptr which accomplishes the same thing

**Pull request merged**